components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/EndPointsApiServiceImpl.java

```diff
@@ -38,9 +44,19 @@ public Response endPointsGet(String apiName, String label, String apiId, Message
         JSONObject responseObj = new JSONObject();
         JSONArray endPointArray = new JSONArray();
         if (gatewayAPIDTO != null) {
-            if (gatewayAPIDTO.getEndpointEntriesToBeAdd() != null) {
+            if (gatewayAPIDTO.getEndpointEntriesToBeAdd() != null ) {
+                EndpointAdminServiceProxy endpointAdminServiceProxy = new EndpointAdminServiceProxy
+                        (gatewayAPIDTO.getTenantDomain());
                 for (GatewayContentDTO gatewayEndpoint : gatewayAPIDTO.getEndpointEntriesToBeAdd()) {
-                    endPointArray.put(gatewayEndpoint.getContent());
+                    try {
+                        if (endpointAdminServiceProxy.isEndpointExist(gatewayEndpoint.getName())) {
+                            endPointArray.put(endpointAdminServiceProxy.getEndpoints(gatewayEndpoint.getName()));
+                        } else {
+                            log.error(gatewayEndpoint.getName() + " was not deployed in the gateway");
```

Comment:
better to have this in the response

Reply:
https://github.com/wso2/carbon-apimgt/pull/8784/commits/2bef8ac27b023cef0271216e1b2942bc739efacc